test/test_bwd.py

```diff
@@ -77,22 +65,28 @@ def run_worker(args):
         device=args.device,
     )
 
+    # Attach to a schedule
+    schedule = PipelineScheduleGPipe(stage, args.chunks, loss_fn=loss_fn)
+
     # Run
     if args.rank == 0:
-        out = stage(x)
+        schedule.step(x)
     elif args.rank == args.world_size - 1:
-        out = stage(y)
+        losses = []
+        out = schedule.step(targets=targets, losses=losses)
     else:
-        stage()
+        schedule.step()
```

Comment:
@H-Huang @wconstab Please check this new UI for training.

Reply:
The return format of `(out, losses)` would seem a bit "arbitrary" to me, meaning, user would need to read closely our doc to know what the format would be. And it could make the API harder to change in future (output is harder to extend/change than argument list).

Reply:
yeah since we are keeping the losses as arguments in the step function, we can just return the a single `out`, makes sense to me!